packages/server/src/api/controllers/datasource.ts

```diff
@@ -127,13 +130,19 @@ export async function verify(
   ctx: UserCtx<VerifyDatasourceRequest, VerifyDatasourceResponse>
 ) {
   const { datasource } = ctx.request.body
-  const existingDatasource = await sdk.datasources.get(datasource._id!)
-
-  const enrichedDatasource = sdk.datasources.mergeConfigs(
-    datasource,
-    existingDatasource
-  )
-
+  let existingDatasource: undefined | Datasource
+  try {
```

Comment:
Do we need the try/catch? Can we not check if the `datasource._id` exists instead?